src/globals.c

```diff
@@ -314,7 +315,12 @@ int set_content_url(char *url)
 		return 0;
 	}
 
-	return set_url(&content_url, url, DEFAULT_CONTENT_URL_PATH);
+	int ret = set_url(&content_url, url, DEFAULT_CONTENT_URL_PATH);
+	if (ret == 0) {
+		content_url_is_local = strncmp(content_url, "file://", 7) == 0;
+	}
```

Comment:
Whilst this obviously works, wouldn't it be better to have an option to set a variable (maybe MAX_TRIES) rather than adding in magic based off the URL? I could see a hypothetical test in the future which was testing the retry facility creating the to-be-downloaded file after a certain amount of time, and I can imagine a user wanting to set a different  MAX_TRIES if their internet was flaky and they wanted to try really hard to download stuff.

Reply:
https://github.com/clearlinux/swupd-client/issues/322

Reply:
And I agree with @icarus-sparry that is a better solution. It is the solution we want to go with to fix the 'make check' problem of taking too long.

Reply:
> Whilst this obviously works, wouldn't it be better to have an option to set a variable (maybe MAX_TRIES) rather than adding in magic based off the URL?

I think by default we shouldn't have retries for local files, and by default we should have retries for remote files. We identify those by the URL scheme. So as I see the highlighted code will exist in some form or another.

About the hypothetical test case: you can still do it if you make the configurable number of retries override any previous existing behavior. So I see a fix #322 complementing the _behavior_ added by this patch -- not in conflict. You can also use a local webserver (in case #322 don't materialize).

If that was not enough, given the size of the change I'd strongly consider just adding it for the sake of 'make check' times.

Reply:
To clarify:

> If that was not enough, given the size of the change I'd strongly consider just adding it for the sake of 'make check' times.

I.e. further improvement to retry logic can easily revert the check and use a different condition instead if that turns out to be the case.

Reply:
Reasonable.